cli/src/main/java/com/devonfw/tools/ide/process/ProcessMode.java

```diff
@@ -12,29 +14,135 @@ public enum ProcessMode {
    * the process remains connected to the terminal so that output is still received. (Only '&' is not used because it just removes awaiting but not sending of
    * SIGHUP. Using nohup would simply result in redirecting output to a nohup.out file.)
    */
-  BACKGROUND,
+  BACKGROUND {
+    @Override
+    public Redirect getRedirectOutput() {
+      return Redirect.INHERIT;
+    }
+
+    @Override
+    public Redirect getRedirectError() {
+      return Redirect.INHERIT;
+    }
+
+    @Override
+    public Redirect getRedirectInput() {
+      return null;
+    }
+  },
   /**
    * Like {@link #BACKGROUND}. Instead of redirecting the output to the parent process, the output is redirected to the 'null file' using
    * {@link ProcessBuilder.Redirect#DISCARD}.
    */
-  BACKGROUND_SILENT,
+  BACKGROUND_SILENT {
+    @Override
+    public Redirect getRedirectOutput() {
+      return Redirect.DISCARD;
+    }
+
+    @Override
+    public Redirect getRedirectError() {
+      return Redirect.DISCARD;
+    }
+
+    @Override
+    public Redirect getRedirectInput() {
+      return null;
+    }
+  },
   /**
    * The process will be started according {@link ProcessBuilder.Redirect#INHERIT} without any detaching of parent process and child process. This setting makes
    * the child process dependant from the parent process! (If you close the parent process the child process will also be terminated.)
    */
-  DEFAULT,
+  DEFAULT {
+    @Override
+    public Redirect getRedirectOutput() {
+      return Redirect.INHERIT;
+    }
+
+    @Override
+    public Redirect getRedirectError() {
+      return Redirect.INHERIT;
+    }
+
+    @Override
+    public Redirect getRedirectInput() {
+      return Redirect.INHERIT;
+    }
+
+  },
   /**
    * The process will be started according {@link ProcessBuilder.Redirect#PIPE} and the standard output and standard error streams will be captured from the
    * parent process. In other words, they will be printed in the console of the parent process. This setting makes the child process dependant from the parent
    * process! (If you close the parent process the child process will also be terminated.)
    */
-  DEFAULT_CAPTURE,
+  DEFAULT_CAPTURE {
+    @Override
+    public Redirect getRedirectOutput() {
+      return Redirect.PIPE;
+    }
+
+    @Override
+    public Redirect getRedirectError() {
+      return Redirect.PIPE;
+    }
+
+    @Override
+    public Redirect getRedirectInput() {
+      return null;
```

Comment:
```suggestion
      return Redirect.INHERIT;
```
Needs to be added or this change gets lost: https://github.com/devonfw/IDEasy/blob/22be644d9662e4979bdb84b7745212a0da114cf1/cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java#L169